adapters/src/main/java/io/realm/RealmCursor.java

```diff
@@ -0,0 +1,688 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import android.annotation.TargetApi;
+import android.content.ContentResolver;
+import android.database.CharArrayBuffer;
+import android.database.ContentObserver;
+import android.database.Cursor;
+import android.database.CursorIndexOutOfBoundsException;
+import android.database.DataSetObserver;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.realm.internal.RealmObjectProxy;
+import io.realm.internal.Row;
+import io.realm.internal.TableOrView;
+
+public class RealmCursor<T extends RealmModel> implements Cursor {
+
+    @Nullable
+    private RealmResults<T> data;
+    @NonNull
+    private final CursorRealmChangeListener changeListener;
+    private int position = -1;
+    private boolean closed;
+
+    @NonNull
+    private final List<ContentObserver> contentObservers;
+    @NonNull
+    private final List<DataSetObserver> dataSetObservers;
+    @NonNull
+    private Bundle extras = Bundle.EMPTY;
+
+    public RealmCursor(@Nullable RealmResults<T> data) {
+        changeListener = new CursorRealmChangeListener();
+        contentObservers = new ArrayList<>();
+        dataSetObservers = new ArrayList<>();
+        setData(data);
+    }
+
+    public void setData(@Nullable RealmResults<T> newData) {
```

Comment:
IMO a Cursor is not a "separate" class, it is just another representation of a `RealmResults`. In the initial PR we talked about having a `RealmResults.asCursor()` method, but for the sake of keeping it seperate I am fine with requiring constructing it with e.g. `new RealmCursor(RealmResults results)`, however being able to replace the result seems like a bad idea in that context.